### PR TITLE
fix: Filter out markdown for og:description

### LINKED ISSUE #6346: Filter out markdown for og:description
Hence we have prerender on board or our ssr-opengraph, would be possible filter out markdown for crawl bots? 

It looks ugly currently and hurts readability when someone shares item in chat

Probably workers issue?

image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6346
- [ ] Requires deployment <snek/rubick/worker>

before:
image

after:

image

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

### <samp>🤖 Generated by Copilot at 93ebc72</samp>

Added a function to detect markdown in strings and used it to remove markdown from collection descriptions in meta tags. This improves SEO by providing cleaner and more relevant metadata for collections.

### <samp>🤖 Generated by Copilot at 93ebc72</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A cunning function, `isMarkdownString`, to discern_
> _The subtle signs of links and styles that mark the words_
> _Of sages who compose their wisdom in markdown._

### CODE REVIEW

**Comment by prury** [comment]
is it supposed to work for the NFT inside the collection aswell?
image

**Comment by vikiival** [comment]
Sir I think this is not correct. I guess @yangwao wanted to strip out markdown stuff like * [] #

**Review by daiagi — APPROVED** [review]

File: utils/markdown.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /\s(__|\*\*)(?!\s)(.(?!\1))+(?!\s(?=\1))/.test(text)
+  )
+}
+
+export const convertMarkdownToText = (text = ''): string => {
+  if (isMarkdownString(text)) {
+    text = text.replaceAll('**', '')
+    text = text.replaceAll('__', '')
+    text = text.replaceAll('\n', ' ')
+    text = text.replaceAll('\n', ' ')
```
Comment (daiagi) [inline utils/markdown.ts:14]:
```suggestion
```

Needless duplicate

File: utils/markdown.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    const urlsMatch = text.match(/\[\S+\]\(https?:\/\/\S+\)/g)
+    if (urlsMatch) {
+      urlsMatch.forEach((match) => {
+        const matchName = match.match(/\[\S+\]/)
+        if (matchName) {
+          text = text.replaceAll(
+            match,
+            matchName[0].slice(1, matchName[0].length - 1)
```
Comment (daiagi) [inline utils/markdown.ts:23]: Not sure what are the requirements
But makes more sense to me to keep the url rather then the display name

**Comment by Jarsen136** [comment]
✅ Updated

before
image


after

image

**Comment by Jarsen136** [comment]
>

> is it supposed to work for the NFT inside the collection aswell? image

Yes, it should also be the same now.

File: utils/markdown.ts (hunk 2 above)
Comment (Jarsen136) [reply utils/markdown.ts:22]: I think the name of the URL would be more readable while sharing the content, at least for this case

image

**Comment by prury** [comment]
@Jarsen136 using a link that directs to the collection is still not showing the description without the markdown: 

image

**Comment by Jarsen136** [comment]
> @Jarsen136 using a link that directs to the collection is still not showing the description without the markdown:
> 
It should be related to the cache. Pls use telegram to test the url sharing again.

**Comment by prury** [comment]
> > @Jarsen136 using a link that directs to the collection is still not showing the description without the markdown:
> 
> It should be related to the cache from discord. Pls use telegram to test the url sharing again. @prury

i had tested on telegram aswell: 

image

**Comment by Jarsen136** [comment]
> > > @Jarsen136 using a link that directs to the collection is still not showing the description without the markdown:
> > 
> > 
> > It should be related to the cache from discord. Pls use telegram to test the url sharing again. @prury
> 
> i had tested on telegram aswell:
> 
> 
> might also be a cache problem, any other place where i can test it?

Yes. It's cache. You could test with this link to refresh the cache. https://deploy-preview-6348--koda-canary.netlify.app/stmn/collection/30?t=2

image

**Comment by prury** [comment]
> 



> > > > @Jarsen136 using a link that directs to the collection is still not showing the description without the markdown:
> > > 
> > > 
> > > It should be related to the cache from discord. Pls use telegram to test the url sharing again. @prury
> > 
> > 
> > i had tested on telegram aswell:
> > might also be a cache problem, any other place where i can test it?
> 
> Yes. It's cache. You could test with this link to refresh the cache. https://deploy-preview-6348--koda-canary.netlify.app/stmn/collection/30?t=2
> image

works fine ty, will just do a general check and i'll change the label

**Review by vikiival — APPROVED** [review]

**Comment by prury** [comment]
oh, it got merged...

image

**Pull request merged**